### ISSUE TITLE
Last modified date does not change

### ISSUE THREAD

**Issue by thomas694** [kapsiR/HaveIBeenPwnedKeePassPlugin#11]
I tested it with v0.6.0 together with KeePass 2.46.

When I run the check, some entries get tagged and even get tag data about how many times it has been used. The database changes to modified state because tags are assigned. But when I save the database all tags disappear.

It's discussable, whether a tag changes the content of a box or not. But the developers of KeePass decided for the former, only items with changed modified date are really saved.

While it may be a feature to assign the HIBP result tags only temporary, why should I bother to assign ignore tags, if they are lost?

**Comment by kapsiR** [comment]
I'll try to reproduce this behavior this weekend when I find some time... 

Can you explain the issue a bit further:
1. Do you use KeePass on Windows?
2. Do you run the check for the whole database?
3. Is it a behavior change with KeePass 2.46 or did you try it with another version?
4. Does this happen all the time?

It is intentional that everything this plugin changes will only be saved if you save the database after changing any state with an action. 

Thanks!

**Comment by thomas694** [comment]
> 1. Do you use KeePass on Windows?
Yes
> 2. Do you run the check for the whole database?
I run it for the whole database and the other time only per item. It makes no difference.
> 3. Is it a behavior change with KeePass 2.46 or did you try it with another version?
I tried the plugin for the first time in this version.
> 4. Does this happen all the time?
It happens all the time and is repeatable.

But, because it's automated and I'm so used to it, I forgot that I have implemented a trigger that automatically synchronizes the database with another location. But for reproducing the problem, the steps can be done manually through the file menu too.

If i check a pawned password the item gets a "pwned" tag, but the item's "Last Modification Time" remains the same. When saving the database, the tag is saved. It can be confirmed by closing and opening the database.
But when this database gets synced with another location, then the databases are synced against each other and the previous state without the tags is restored.
Only when I make an additional change to the item and of course the "Last Modification Time" changes, then the tag is still there after saving the database and syncing it with the other location.

But of course no one makes an additional manual change to every item after doing a check of all passwords automatically.
I don't know which steps the program makes, when assigning a tag through the menu, but then the item gets a new "Last Modification Time" and the tag remains after syncing.

**Comment by kapsiR** [comment]
Thanks for the detailed explanation! 

I'll try to reproduce this behavior for myself and give an update accordingly. 

I assume, I can fix this within the next few days.

**Comment by kapsiR** [comment]
@thomas694 I've implemented a new option to enable the date changes with the [new preview release](https://github.com/kapsiR/HaveIBeenPwnedKeePassPlugin/releases/tag/v0.7.0-preview).

I think, this behavior should explicitly be enabled, so I implemented this as new feature instead of a fix...

**Comment by thomas694** [comment]
Well, I know now which option to toggle to make it behave correctly. Thanks.

But I think without enabling the new option it's not the expected behavior when I use the program as a whole without considering that some changes come from a plugin I installed. When a tag is added through the original program (eg. through the tag submenu) the item gets a new last modified date and (or because of that) the item's history is advanced. I think, if a plugin adds a tag the outcome should be the same.

I just found another thing. This time I forgot to disable the option "check password when entry gets modified". When validating and opening an item for viewing only, after closing the dialog (with cancel) it still sends a request and shows the pwned result message.

All tested in a clean default database.
